app/src/main/java/com/github/se/travelpouch/ui/dashboard/AddActivityScreen.kt

```diff
@@ -0,0 +1,203 @@
+import androidx.core.text.isDigitsOnly
+import com.github.se.travelpouch.model.Location
+import com.github.se.travelpouch.model.activity.Activity
+import com.github.se.travelpouch.model.activity.ActivityModelView
+import com.google.firebase.Timestamp
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.GregorianCalendar
+import java.util.Locale
+
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@Composable
+fun AddActivityScreen(activityModelView: ActivityModelView) {
+  val dateFormat =
+      SimpleDateFormat("dd/MM/yyyy", Locale.getDefault()).apply {
+        isLenient = false // strict date format
+      }
+
+  val gregorianCalendar = GregorianCalendar()
+
+  var title by remember { mutableStateOf("") }
+  var description by remember { mutableStateOf("") }
+  var dateText by remember { mutableStateOf("") }
+  var location by remember { mutableStateOf("Placeholder") }
+
+  val context = LocalContext.current
+
+  val placeholerLocation = Location(0.0, 0.0, Timestamp(0, 0), "name")
+
+  Scaffold(modifier = Modifier.testTag("AddActivityScreen")) {
+    Column(
+        modifier = Modifier.fillMaxSize().padding(16.dp).verticalScroll(rememberScrollState()),
+        verticalArrangement = Arrangement.spacedBy(16.dp)) {
+          OutlinedTextField(
+              value = title,
+              onValueChange = { title = it },
+              label = { Text("Title") },
+              placeholder = { Text("Title") },
+              modifier = Modifier.testTag("titleField"))
+
+          OutlinedTextField(
+              value = description,
+              onValueChange = { description = it },
+              label = { Text("Description") },
+              placeholder = { Text("Description") },
+              modifier = Modifier.testTag("descriptionField"))
+
+          OutlinedTextField(
+              value = dateText,
+              onValueChange = {
+                if (it.isDigitsOnly() && it.length <= 8) {
+                  dateText = it
+                }
+              },
+              label = { Text("Date") },
+              placeholder = { Text("01/01/1970") },
+              visualTransformation = DateVisualTransformation(),
+              keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
+              modifier = Modifier.testTag("dateField"))
+
+          OutlinedTextField(
+              value = location,
+              onValueChange = {},
+              label = { Text("Location") },
+              placeholder = { Text("Location") },
+              modifier = Modifier.testTag("locationField"))
+
+          Button(
+              enabled =
+                  location.isNotBlank() &&
+                      title.isNotBlank() &&
+                      description.isNotBlank() &&
+                      dateText.isNotBlank(),
+              onClick = {
+                try {
+                  val finalDate = convertStringToDate(dateText, dateFormat, gregorianCalendar)
```

Comment:
I mentioned that the current approach for handling dates is fine, but it might be even better to use a function like a DatePicker to further improve the user experience.

Reply:
Ok, we can imlpement for a later sprint. But yes, it may give a better user experience